src/store/sagas/index.ts

```diff
@@ -38,3 +39,9 @@ export function* watchAuthSagas() {
   yield takeEvery(actionTypes.LOGIN, login);
   yield takeEvery(actionTypes.LOGOUT, logout);
 }
+
+export function* watchWidgetConfigs() {
+  yield takeEvery(actionTypes.SET_ASSET, machineChanged);
+  yield takeEvery(actionTypes.SET_NEW_WIDGET, addWidget);
+  yield pollSyncWidgetConf();
+}
```

Comment:
Since the watcher is running in background, how do you make sure that save config saga will have access to correct config when saving? When machine is changed, will there be a possibility of config changing to new asset selected before old is synced? 
Also, why do you have to handle addWidget like this?